### ISSUE TITLE
Jupyter Notebooks not working

### ISSUE THREAD

**Issue by benz0li** [coder/code-server#3782]
## OS/Web Information

- Web Browser: Firefox 89.0.2 (64-bit)
- Local OS: macOS 10.14.6 (Mojave)
- Remote OS: [Debian buster](https://hub.docker.com/_/debian)
- Remote Architecture: 64-bit
- `code-server --version`: 3.11.0 4e8cd09ef0412dfc7b148b7639a692e20e4fd6dd

## Steps to Reproduce

* Create a `.ipynb` file 

## Expected

Jupyter extension should load and display the Jupyter notebook.

## Actual

![jupyter_error](https://user-images.githubusercontent.com/20233085/125745202-0a0e1c09-3c26-48e5-9ba3-721e0a9cc7bc.png)

## Logs

_Log (Remote Extension Host)_

```
[2021-07-15 07:06:36.925] [exthost] [error] Activating extension ms-toolsai.jupyter failed due to an error:
[2021-07-15 07:06:36.925] [exthost] [error] Error: Cannot find module '/home/jovyan/.local/share/code-server/extensions/ms-toolsai.jupyter-2021.6.99/out/client/extension'
Require stack:
- /opt/code-server/lib/vscode/out/vs/loader.js
- /opt/code-server/lib/vscode/out/bootstrap-amd.js
- /opt/code-server/lib/vscode/out/bootstrap-fork.js
	at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
	at Function.Module._load (internal/modules/cjs/loader.js:746:27)
	at Function.n._load (/opt/code-server/lib/vscode/out/vs/workbench/services/extensions/node/extensionHostProcess.js:103:27915)
	at Function.E._load (/opt/code-server/lib/vscode/out/vs/workbench/services/extensions/node/extensionHostProcess.js:103:24661)
	at Function.c._load (/opt/code-server/lib/vscode/out/vs/workbench/services/extensions/node/extensionHostProcess.js:93:59193)
	at Module.require (internal/modules/cjs/loader.js:974:19)
	at require (internal/modules/cjs/helpers.js:92:18)
	at Function.t [as __$__nodeRequire] (/opt/code-server/lib/vscode/out/vs/loader.js:5:101)
	at f._loadCommonJSModule (/opt/code-server/lib/vscode/out/vs/workbench/services/extensions/node/extensionHostProcess.js:103:26067)
	at f._doActivateExtension (/opt/code-server/lib/vscode/out/vs/workbench/services/extensions/node/extensionHostProcess.js:89:12599)
	at f._activateExtension (/opt/code-server/lib/vscode/out/vs/workbench/services/extensions/node/extensionHostProcess.js:89:11658)
	at Object.actualActivateExtension (/opt/code-server/lib/vscode/out/vs/workbench/services/extensions/node/extensionHostProcess.js:89:9243)
	at l._activateExtension (/opt/code-server/lib/vscode/out/vs/workbench/services/extensions/node/extensionHostProcess.js:84:85443)
	at /opt/code-server/lib/vscode/out/vs/workbench/services/extensions/node/extensionHostProcess.js:84:85125
	at Array.map (<anonymous>)
	at l._activateExtensions (/opt/code-server/lib/vscode/out/vs/workbench/services/extensions/node/extensionHostProcess.js:84:85113)
	at l.activateByEvent (/opt/code-server/lib/vscode/out/vs/workbench/services/extensions/node/extensionHostProcess.js:84:82447)
	at f._activateByEvent (/opt/code-server/lib/vscode/out/vs/workbench/services/extensions/node/extensionHostProcess.js:89:10142)
	at /opt/code-server/lib/vscode/out/vs/workbench/services/extensions/node/extensionHostProcess.js:89:20014
	at processTicksAndRejections (internal/process/task_queues.js:95:5)
```

## Screenshot

See above.

## Notes

This issue can be reproduced in VS Code: No

---

@jsjoeio This is the bug report you asked for in issue https://github.com/cdr/code-server/issues/3776.

**Comment by jsjoeio** [comment]
Planning to look into this after we wrap up 3.11!

**Comment by benz0li** [comment]
FYI It worked fine with code-server 3.10.2 387b12ef4ca404ffd39d84834e1f0776e9e3c005, Python 2021.5.926500501 and Jupyter 2021.6.832593372.

**Comment by gogobd** [comment]
I'm not sure if this is related, but in 3.11 I get an error when calling "Jupyter: Create New Blank Notebook": 
<img width="521" alt="Screenshot 2021-07-31 at 15 29 23" src="https://user-images.githubusercontent.com/728976/127741432-6f94b395-98cb-4a53-8d58-3c85e75251a9.png">

**Comment by jsjoeio** [comment]
Thanks for mentioning @gogobd - I got that same here and we're tracking in this issue: https://github.com/cdr/code-server/issues/3874

**Comment by adalseno** [comment]
Same issue for me with my installation on my iMac with Os X 10.14 using Homebrew (the latest version of code-server is 3.12.0); jupyter files won't open:

`code-server: v3.12.0
VS Code: v1.60.0
Commit: 4cd55f94c0a72f05c18cea070e10b969996614d2
Date: 2021-09-15T19:34:37Z (1 mo ago)
Browser: Mozilla/5.0 (X11; CrOS x86_64 14150.64.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.104 Safari/537.36`

Python extension installed (v2021.10.1365161279 from open-vsx) and Jupyter too (v2021.8.12; actually I tried to install a more recent version using open-vsx, the installer said that version 2021.10.100 was succesfully installed code-server shows me the other version). They are no python or jupyter duplicated versions in the extension directory.

I can select the python interpreter, also for jupyter server, but when I try to open a .ipynb file I get the following error:

![Screenshot 2021-11-02 00 18 05](https://user-images.githubusercontent.com/2405291/139754443-7502b38f-0bcb-4374-9f59-212d3525435c.png)
![Screenshot 2021-11-02 00 22 46](https://user-images.githubusercontent.com/2405291/139754873-07f5b6c8-6b85-40ac-85fd-6c17adcf6e37.png)

The same issue if I run from the iMac where code-server is installed or remotely using my Chromebook.
In local Vscode on my iMac jupyter files open without any issue selecting the same python interpreter.

PS (off-topic) adding the suggested lines in prdouct.json to use open-vsx does not seem to work for me even after a server restart:

`
            "extensionsGallery": {
              "serviceUrl": "https://open-vsx.org/vscode/gallery",
              "itemUrl": "https://open-vsx.org/vscode/item"
            }
`

**Comment by adalseno** [comment]
Just a little update. I created a Docker container using the command from the FAQ slightly modified (I use port 8081 and a different local config dir since I have code-server running on port 8080:
```
docker run -d --name code-server -p 8081:8080 \
  -v "$HOME/.config_docker:/home/coder/.config" \
  -v "$PWD/code-projects/:/home/coder/project" \
  -u "$(id -u):$(id -g)" \
  -e "DOCKER_USER=$USER" \
  codercom/code-server:latest
```
then installed the python extension (that installs also jupyter one), but the issue is still there.  Installing miniconda and/or jupyter does not change the situation
![Schermata 2021-11-03 alle 12 35 22](https://user-images.githubusercontent.com/2405291/140074830-410aeeee-2332-44b1-a351-fb21b87c0336.png)
.
 PS No errors in the docker log and the server is running on https without any issue, simply it won't open jupyter notebooks or create new ones

**Comment by jsjoeio** [comment]
@adalseno I'm not convinced you're actually installing the extensions from OpenVSX. Can you please follow the docs and try [this](https://coder.com/docs/code-server/latest/FAQ#how-do-i-install-an-extension-manually)? Let us know if that works

**Comment by adalseno** [comment]
Hi @jsjoeio , thank you very much for your kind reply.
I did like this:
`SERVICE_URL=https://open-vsx.org/vscode/gallery ITEM_URL=https://open-vsx.org/vscode/item code-server --install-extension ms-python.python`
Am I doing wrong? 
But the error is still there (I cleaned the extension dir before installing).
![Schermata 2021-11-03 alle 19 16 04](https://user-images.githubusercontent.com/2405291/140169237-91fb77df-93a9-482c-8b19-1ed37610c72d.png)

**Comment by jsjoeio** [comment]
@adalseno thanks for the quick reply - nope that's exactly right so something is not right!

**Comment by adalseno** [comment]
Hi @jsjoeio ,I don't know if this can help, but, after uninstalling the extensions and cleaning the extension dir, I tried to download the extensions from vsx and manually install them via command line. I got en error installing the latest version of ms-toolsai since it says it is not compatible with Vscode 1.60.0, see the screenshot:
![Schermata 2021-11-03 alle 20 13 26](https://user-images.githubusercontent.com/2405291/140177659-7b069160-6fd0-4f6d-a2fa-027f94a86942.png)
Anyway, even with manual installation the error is still there :(

**Comment by jsjoeio** [comment]
Yeah, we're in the process of updating VS Code so it should be compatible in the next release 🤞 (sorry for the issues!)

**Comment by adalseno** [comment]
Hi @jsjoeio , will it work if, in the meanwhile, I install a previous version of the plugins? And if so which version? Thanks

**Comment by jsjoeio** [comment]
@adalseno you might find the answer to that [here](https://github.com/cdr/code-server/issues/4333)!

**Comment by adalseno** [comment]
Yes, it worked for 3.9.3 but doing the same on 3.12.0 does not work :
![Schermata 2021-11-04 alle 17 46 29](https://user-images.githubusercontent.com/2405291/140384645-044052d3-9ae7-4972-8a9b-aa2ef7ae8dab.png)
Unfortunately there is no brew formula for version 3.9.3. I hope you will be able to fix 3.12 soon.
Thanks.

**Comment by jsjoeio** [comment]
Dang it! Thanks for the report. 

> Unfortunately there is no brew formula for version 3.9.3

Yes, you're right :( I don't think brew keeps old versions around. You can install code-server through npm/yarn and use npm. i.e. 

```shell
yarn global add code-server@3.9.3
```

**Comment by ndgayan** [comment]
This worked
[https://github.com/cdr/code-server/issues/3874#issuecomment-923383398](url)

**Comment by ndgayan** [comment]
This worked

https://github.com/cdr/code-server/issues/3874#issuecomment-923383398

**Comment by adalseno** [comment]
Hi ndgayan , thank you very much, it worked perfectly on my iMac.
To recap:
I installed code-server using brew but it had issues with jupyter notebooks. So, as suggested by @ndgayan , I installed code-server as a standalone with:
`curl -fsSL https://code-server.dev/install.sh | sh -s -- --method=standalone`
modified my plist to point to the new version `~/.local/lib/code-server-3.12.0/code-server` and everything worked perfectly (no need to reinstall any extension, I already downloaded them from open-vsx,  or modify the config files)
It's a good workaround waiting for 3.12.1
![Schermata 2021-11-06 alle 13 43 52](https://user-images.githubusercontent.com/2405291/140610102-47bbbe53-9e7e-407d-8ca8-33249a1edc4f.png)

**Comment by fritterhoff** [comment]
Using the new version 4.0.0 I get quite strange errors, telling me that I must enable a proposed api:
```
[Extension Host] extension activation failed Error: Extension 'ms-toolsai.jupyter' CANNOT use API proposal: notebookEditor.
Its package.json#enabledApiProposals-property declares: [] but NOT notebookEditor.
 The missing proposal MUST be added and you must start in extension development mode or use the following command line switch: --enable-proposed-api ms-toolsai.jupyter
```

**Comment by benz0li** [comment]
Same problem as @fritterhoff with my setup (https://vscode-r.jupyter.b-data.ch).

**Comment by jsjoeio** [comment]
I'm testing this out with the v4.0.0 pre-release. I'll update this comment with my notes.

## Notes

The file loads but the Jupyter extension is complaining.

```javascript
interactive/cell/title is a proposed menu identifier. It requires 'package.json#enabledApiProposals: ["notebookEditor"]' and 

is only available when running out of dev or with the following command line switch: --enable-proposed-api ms-toolsai.jupyter
```

**Comment by fritterhoff** [comment]
> `./code-server-4.0.0-macos-amd64/bin/code-server --enable-proposed-api "notebookEditor"`

Shouldn't that be `./code-server-4.0.0-macos-amd64/bin/code-server --enable-proposed-api "ms-toolsai.jupyter"`?

**Comment by fritterhoff** [comment]
@jsjoeio could it even be that the issue is caused by the extension? As the log says:
`Its package.json#enabledApiProposals-property declares: [] but NOT notebookEditor.`

So looking into the extensions vsix and the package.json proofs that there is no `enabledApiProposals` property at all.

**Comment by fritterhoff** [comment]
Using the latest version (2022.1.1001585725) provides:
```
    "enabledApiProposals": [
        "notebookConcatTextDocument",
        "notebookControllerKind",
        "notebookDebugOptions",
        "notebookDeprecated",
        "notebookEditor",
        "notebookEditorDecorationType",
        "notebookEditorEdit",
        "notebookMessaging",
        "notebookMime",
        "notebookCellExecutionState",
        "portsAttributes",
        "textDocumentNotebook",
        "quickPickSortByLabel"
    ],
```

**Comment by fritterhoff** [comment]
> @jsjoeio could it even be that the issue is caused by the extension? As the log says: `Its package.json#enabledApiProposals-property declares: [] but NOT notebookEditor.`
> 
> So looking into the extensions vsix (version ms-toolsai.jupyter-2021.11.1001552333) and the package.json proofs that there is no `enabledApiProposals` property at all.

Digging a little bit further in this approach leads me to an other point

By default this is done:
```
/** @coder Always enable the proposed API. */
this._envEnablesProposedApiForAll = true;
this._logService.debug('[Extensions]', 'Coder: Enabled All Proposed APIs');
```

So from my point of view we never should run into: https://github.com/coder/vscode/blob/main/src/vs/workbench/services/extensions/common/abstractExtensionService.ts#L1215

**Comment by benz0li** [comment]
Related to https://github.com/coder/code-server/issues/4480

**Comment by jsjoeio** [comment]
@fritterhoff thanks for sharing your notes too!

> So from my point of view we never should run into: https://github.com/coder/vscode/blob/main/src/vs/workbench/services/extensions/common/abstractExtensionService.ts#L1215

Yes! @code-asher said the same thing yesterday so it's very very weird. Did you get the same errors with the latest extension(2022.1.1001585725)? I must have an out of date one. I'll see if I can try that.

Nice find @benz0li!

**Comment by fritterhoff** [comment]
2022.1.1001585725 does not work due to mismatch in the code version

**Comment by jsjoeio** [comment]
> not work due to mismatch in the code version

VS Code version? I wonder if I edit source and make it compatible if that will work.

**Comment by jsjoeio** [comment]
Here are notes as part of my second time investigating this.

## Notes

### Test with 2022.1.1001585725 (no modifications

1. Download from Open VSX [here](https://open-vsx.org/extension/ms-toolsai/jupyter)
2. Install in code-server (v4.0.0 pre-release)

Expected: it installs
Actual: it fails due to mismatch like @fritterhoff pointed out.
![image](https://user-images.githubusercontent.com/3806031/146421132-60b4d964-03a2-4580-b4a9-c1726c14c33a.png)

You can see the latest version of VS code required [here](https://github.com/microsoft/vscode-jupyter/blob/main/package.json#L29-L31). 

We can also inspect the source code ourselves by changing the downloaded `.vsix` file to a `.zip` and then unpacking. 

![image](https://user-images.githubusercontent.com/3806031/146421415-865531e0-03fe-4260-bef5-31a0e1dc384b.png)

Opening that folder (called `extension`) in code-server to inspect the files, we see this. 

![image](https://user-images.githubusercontent.com/3806031/146421616-a8bb4aa3-aee5-4d72-aea0-117ee7ed29ad.png)

One theory to test is to change this version to match the one in code-server: `1.63.0` and then zip, change to `.vsix` and try installing.

**Comment by fritterhoff** [comment]
All in all that does not sound good ...

**Comment by jsjoeio** [comment]
I'm going to look into this further and start with [this block](https://github.com/coder/vscode/blob/main/src/vs/workbench/services/extensions/common/abstractExtensionService.ts#L1207-L1211) seeing if this actually runs (with the latest version of the Jupyter extension that will install).

**Comment by benz0li** [comment]
The Jupyter v2021.8.12 and Python v2021.10.1365161279 extension worked well with code-server v3.12.0 (aka VS Code v1.60.0). You may check on https://demo.jupyter.b-data.ch; login with GitHub account.

**Comment by jsjoeio** [comment]
> All in all that does not sound good ...

I know :( There is a high chance though something is not working in code-server though. This is just a hard extension to debug as you've experienced as well.

The other thing we could test is seeing if the latest version works in [openvscode-server](https://github.com/gitpod-io/openvscode-server).

**Comment by fritterhoff** [comment]
> The Jupyter v2021.8.12 and Python v2021.10.1365161279 extension worked well with code-server v3.12.0 (aka VS Code v1.60.0). You may check on https://demo.jupyter.b-data.ch; login with GitHub account.

Yeah atm we also run the "old" jupyter extension, but the new brings some great features we would love to use.

**Comment by jsjoeio** [comment]
> The Jupyter v2021.8.12 and Python v2021.10.1365161279 extension worked well with code-server v3.12.0 (aka VS Code v1.60.0). You may check on https://demo.jupyter.b-data.ch; login with GitHub account.

Yeah, this works great! 

![image](https://user-images.githubusercontent.com/3806031/146432073-a8a13e0c-7b80-4e9e-8962-e2b95837e0b0.png)

Thanks for posting the versions as well. I could also test those same versions with 4.0.0

**Comment by fritterhoff** [comment]
Those also work (from my tests) with 4.0.0

**Comment by jsjoeio** [comment]
> Those also work (from my tests) with 4.0.0

Oh saved me some time! Thanks for sharing that.

Hmm...I guess this is the logical next step:

> I'm going to look into this further and start with this block seeing if this actually runs (with the latest version of the Jupyter extension that will install).

**Comment by benz0li** [comment]
@jsjoeio I may force install Jupyter v2021.8.12 and Python v2021.10.1365161279 for code-server v4.0.0 at https://vscode-r.jupyter.b-data.ch if you wish.

I don't need the latest versions of these extension. I'm happy to pin older ones, if they work.

**Comment by fritterhoff** [comment]
> I'm going to look into this further and start with [this block](https://github.com/coder/vscode/blob/main/src/vs/workbench/services/extensions/common/abstractExtensionService.ts#L1207-L1211) seeing if this actually runs (with the latest version of the Jupyter extension that will install).

I think that would make the most sense.

> I don't need the latest versions of these extension. I'm happy to pin older versions of these extension, if they work.

Well it would be sweet if it works ;)

**Comment by jsjoeio** [comment]
> @jsjoeio I may force install Jupyter v2021.8.12 and Python v2021.10.1365161279 for code-server v4.0.0 at https://vscode-r.jupyter.b-data.ch if you wish.

I'd be curious to see if that worked as well 👀

**Comment by benz0li** [comment]
> Well it would be sweet if it works ;)

> I'd be curious to see if that worked as well 👀

It does not.

@fritterhoff I have whitelisted you at https://vscode-r.jupyter.b-data.ch, too.

---

Going to rollback in 90 mins.

**Comment by fritterhoff** [comment]
Thats strange... I'll give it another try

**Comment by fritterhoff** [comment]
Ok... strange I thought that worked yesterday. Does not work for me. Sorry.

**Comment by jsjoeio** [comment]
Investigation Session (3)

---

## Notes

### Looking at the Jupyter extension in the VS Code Marketplace

[This one](https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter) does have the `enabledApiProposals` in their `package.json`:

```json
    "enabledApiProposals": [
        "notebookConcatTextDocument",
        "notebookControllerKind",
        "notebookDebugOptions",
        "notebookDeprecated",
        "notebookEditor",
        "notebookEditorDecorationType",
        "notebookEditorEdit",
        "notebookMessaging",
        "notebookMime",
        "notebookCellExecutionState",
        "portsAttributes",
        "textDocumentNotebook",
        "quickPickSortByLabel"
    ],
```

but this requires Insiders and won't work in vanilla VS Code.

### Looking at the extension in vanilla VS Code

When you open vanilla VS Code and search for Jupyter, it installs v2021.11.1001550889. Strangely enough, this exact version is not available on Open VSX 🤔 The closest I imagine is 2021.11.1001552333.

I'll try that in code-server.

**Comment by benz0li** [comment]
@jsjoeio I've rolled back.

FYI Jupyter v2021.11.1001552333 ist the version deployed at https://vscode-r.jupyter.b-data.ch.

**Comment by benz0li** [comment]
@jsjoeio Nonetheless, the issue rather lies with code-server than the extension(s).

Usually, older versions of the extension(s) work with newer versions of code-server.

**Comment by jsjoeio** [comment]
I will submit a PR into `coder/vscode` shortly

**Comment by fritterhoff** [comment]
@jsjoeio Hi, I just tried the latest release artifacts as they also update the vscode version(?)

Sadly: The issue seems not to be sovled. Still getting the same issue.

**Comment by fritterhoff** [comment]
I think we should open this issue again, as also the Python Extension throws a similar error.
![grafik](https://user-images.githubusercontent.com/32478819/146641482-292baf5d-080c-4520-b4a8-ac2e05fb2c99.png)

**Comment by benz0li** [comment]
> @jsjoeio Hi, I just tried the latest release artifacts as they also update the vscode version(?)

@fritterhoff The latest release artefacts do not include the fix.  
ℹ️ See https://github.com/coder/code-server/blob/main/vendor/package.json#L10 and https://github.com/coder/vscode/commits/main.

**Comment by fritterhoff** [comment]
Ahhhh fu ...

**Comment by fritterhoff** [comment]
@benz0li FYI: I've built my own version - well the extensions work, but some other rather odd effects occure. 

Prop. you also will recognize the one or the other (strange) behavior.

**Comment by fritterhoff** [comment]
So far I figured out at least two issues that are rather strange:

1. Running NBs does not work: (According to browser console File not found)

https://user-images.githubusercontent.com/32478819/146665568-7ea1ed7e-8608-41fe-b6f2-7b7d89c9cc47.mp4

2. Changing Python Interpreter does not work. `"Failed to set 'pythonPath'. Error: Unable to write to Folder Settings because no resource is provided."`

https://user-images.githubusercontent.com/32478819/146665708-4a95dbc2-c678-4dd6-bbcc-42aacdafc505.mp4

**Comment by code-asher** [comment]
Thank you for testing it and nice find!  If it works with `?folder` I
think I might know where the problem is.  I will take a look soon.

**Comment by fritterhoff** [comment]
> Thank you for testing it and nice find! If it works with `?folder` I think I might know where the problem is. I will take a look soon.

Yes, adding the param `?folder=` works fine. Afterwards the new version 4.0.0 works great in my case! Great tool for plenty use cases!

**Comment by bhatiaabhinav** [comment]
I found this thread after hours of searching on the Internet for the enable-proposed-api problem. And it took another hour to read and understand.

Here is what worked for me finally (@jsjoeio's solution https://github.com/coder/code-server/issues/3782#issuecomment-996202105) :
1. Install v4.0.0 prelease code-server.
2. Install the latest python extension from the sidebar (and the Jupyter extension will get installed automatically). In my case it was Python v2021.12.1559732655 (and Jupyter v2021.11.1001552333).
3. Add the following snippet to `~/.local/share/code-server/extensions/ms-toolsai.jupyter-\<version\>`
```json
"enabledApiProposals": [
        "notebookConcatTextDocument",
        "notebookControllerKind",
        "notebookDebugOptions",
        "notebookDeprecated",
        "notebookEditor",
        "notebookEditorDecorationType",
        "notebookEditorEdit",
        "notebookMessaging",
        "notebookMime",
        "notebookCellExecutionState",
        "portsAttributes",
        "textDocumentNotebook",
        "quickPickSortByLabel"
    ],
```
Start `code-server` without any arguments.

**Comment by jsjoeio** [comment]
@bhatiaabhinav sorry you didn't find this sooner (and sorry for the long read)! 

Thanks for taking the time to post your quick 3-step solution. Hoping it saves someone else time later!

**Comment by benz0li** [comment]
@jsjoeio Not working with code-server v4.0.1. Tested with the [latest release artefacts](https://github.com/coder/code-server/actions/runs/1655552758).

@code-asher Please check with the deployment at https://vscode-r.jupyter.b-data.ch.  
ℹ️ There is code-server v4.0.1 with extensions Python v2021.12.1559732655 and Jupyter v2021.11.1001552333 installed.

**Comment by fritterhoff** [comment]
@benz0li from my point of view it seams to work in your installation after putting the `/?folder=/home/jovyan` in the address.

**Comment by fritterhoff** [comment]
> Output in _web console_:
> 
> ```
> [Error] ReferenceError: Can't find variable: J
> 	i (notebookWebviewPreloads.js:3)
> 	Module Code (notebookWebviewPreloads.js:3:24751)
> 	evaluate
> 	moduleEvaluation
> 	(anonymous function)
> 	promiseReactionJob
> ```

Jep seeing it finally ...

**Comment by fritterhoff** [comment]
Yep. Getting the same error.

**Comment by fritterhoff** [comment]
So I'm playing arround with different versions. Until now I can confirm https://github.com/coder/vscode/commit/0e2dc9d8cffda388b1f8646b4be881d043ad8ed7 is working, the current artifacts are broken and (try and error) attempt using https://github.com/coder/vscode/commit/acfcdb5a3896a125d92bbc36c7dcaf045e8e9dbb also does not work. Atm I'm building a version using https://github.com/coder/vscode/commit/53bc7512d7af7105fc5908e07eed9f9a891e94d6 so the issue can be isolated to one or two commits ...

**Comment by fritterhoff** [comment]
So, it seems like either https://github.com/coder/vscode/commit/53bc7512d7af7105fc5908e07eed9f9a891e94d6 or https://github.com/coder/vscode/commit/d50b73c41535a688e4bace787a1f5d829d26af40 causes the issues. 

As https://github.com/coder/vscode/commit/53bc7512d7af7105fc5908e07eed9f9a891e94d6 "only" adds a missing file I'm pretty sure, that https://github.com/coder/vscode/commit/d50b73c41535a688e4bace787a1f5d829d26af40 (https://github.com/coder/vscode/pull/23) causes the issues. Well now it's up to you @code-asher @jsjoeio to dig into that problems and maybe to revert https://github.com/coder/vscode/pull/23?

**Comment by jsjoeio** [comment]
@benz0li @fritterhoff wow - thank you both for digging in! 

@code-asher should we revert https://github.com/coder/vscode/commit/d50b73c41535a688e4bace787a1f5d829d26af40 then?

**Comment by jsjoeio** [comment]
@fritterhoff you rock! I'm running it locally with [this revert](https://github.com/coder/vscode/pull/40) to ensure it works as expected too!

**Comment by jsjoeio** [comment]
That seemed to do it! Thank you again for making this very easy to fix. We couldn't have done it this fast without both of you


https://user-images.githubusercontent.com/3806031/148261234-ba91f181-5f18-4d7e-a693-ec03e7dfc73b.mov

**Comment by fritterhoff** [comment]
Sorry a little bit off topic: 
@benz0li may I ask how you manage to install the extensions as built in extensions in your docker containers? I've got a similar use case and did not find a solution until now.

**Comment by benz0li** [comment]
> @benz0li may I ask how you manage to install the extensions as built in extensions in your docker containers? I've got a similar use case and did not find a solution until now.

e.g. `code-server --extensions-dir $CODE_BUILTIN_EXTENSIONS_DIR --install-extension ms-toolsai.jupyter` with `CODE_BUILTIN_EXTENSIONS_DIR=/opt/code-server/vendor/modules/code-oss-dev/extensions`;`PATH=/opt/code-server/bin:$PATH`.

**Comment by taeuscherpferd** [comment]
Sorry for posting in a closed thread, but this doesn't appear to be working in the npm install of '4.1.0'. Checking the web browser revealed the same error that you showed above about the ipynbMain.js file missing. I would bet that this is a really simple fix of changing a reference or something for the npm install, but I'm not familiar enough with the codebase to even know where to look. I was able to get around the issue by pulling the compiled version of ipynbMain.js from my install of vscode, and placing it in the location that code-server was looking at. (I also had to create a dist directory for it). Hopefully this gives enough information to help anyone else looking to get around this, as well as helps the maintainer easily resolve this 😄

**Comment by code-asher** [comment]
Thank you for reporting!

I verified the ipynb extension is missing the dist directory after
installing (`yarn global add code-server` in my case).

It exists in the standalone release though so I guess it is unlikely to
be a build issue.

I downloaded the npm package artifact from CI but it looks fine so not a
packaging issue either.

Which I guess means something went wrong when we ran `yarn publish`.  I
think we will need to try re-publishing, maybe as 4.1.0-1 although I am
nervous the same thing will happen.
